### PR TITLE
SophosEP Data Connector

### PR DESCRIPTION
**Pull request by vu-socprime**
## Proposed Changes

  - add SophosEP Data Connector

### CODE REVIEW

File: Solutions/SophosEP/Data Connectors/SophosEP_API_FunctionApp.json [hunk 1]
```diff
@@ -0,0 +1,122 @@
+{
+	"id": "SophosEP",
+	"title": "Sophos Endpoint Protection",
+	"publisher": "Sophos",
+	"descriptionMarkdown": "The [Sophos Endpoint Protection](https://www.sophos.com/en-us/products/endpoint-antivirus.aspx) data connector provides the capability to ingest [Sophos events](https://docs.sophos.com/central/Customer/help/en-us/central/Customer/common/concepts/Events.html) into Azure Sentinel. Refer to [Sophos Central Admin documentation](https://docs.sophos.com/central/Customer/help/en-us/central/Customer/concepts/Logs.html) for more information.",
+	"additionalRequirementBanner": "This data connector depends on a parser based on a Kusto Function to work as expected [**SophosEPEvent**](https://aka.ms/sentinel-SophosEP-parser) which is deployed with the Azure Sentinel Solution.",
```
Comment (v-admahe) [inline Solutions/SophosEP/Data Connectors/SophosEP_API_FunctionApp.json:6]: As per the latest template there is change in additional requirement banner text could yo please update it. You can take a reference from [here](https://github.com/Azure/Azure-Sentinel/blob/master/DataConnectors/Templates/Connector_REST_API_AzureFunctionApp_template/DataConnector_API_AzureFunctionApp_template.json)

Thanks,
Aditi Maheshwari

File: Solutions/SophosEP/Data Connectors/SophosEP_API_FunctionApp.json [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+	"instructionSteps": [{
+			"title": "",
+			"description": ">**NOTE:** This connector uses Azure Functions to connect to the Sophos Central APIs to pull its logs into Azure Sentinel. This might result in additional data ingestion costs. Check the [Azure Functions pricing page](https://azure.microsoft.com/pricing/details/functions/) for details."
+		},
+		{
+			"title": "",
+			"description": ">**(Optional Step)** Securely store workspace and API authorization key(s) or token(s) in Azure Key Vault. Azure Key Vault provides a secure mechanism to store and retrieve key values. [Follow these instructions](https://docs.microsoft.com/azure/app-service/app-service-key-vault-references) to use Azure Key Vault with an Azure Function App."
+		},
+		{
+			"description": ">**NOTE:** This data connector depends on a parser based on a Kusto Function to work as expected [**SophosEPEvent**](https://aka.ms/sentinel-SophosEP-parser) which is deployed with the Azure Sentinel Solution."
```
Comment (v-admahe) [inline Solutions/SophosEP/Data Connectors/SophosEP_API_FunctionApp.json:76]: As per the latest template there is change in additional requirement banner text so we have to update here as well could you please update it. You can take a reference from [here](https://github.com/Azure/Azure-Sentinel/blob/master/DataConnectors/Templates/Connector_REST_API_AzureFunctionApp_template/DataConnector_API_AzureFunctionApp_template.json)

Thanks,
Aditi Maheshwari

**Review by v-admahe — CHANGES_REQUESTED** [review]

File: Solutions/SophosEP/Parsers/SophosEPEvent.txt [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+| extend EventVendor = 'Sophos'
+| extend EventProduct = 'Endpoint Protection'
+| project-rename DstUserSid=user_id_s,
+    CustomerId=customer_id_g,
+    EventSeverity=severity_s,
+    Created=created_at_t,
+    SrcIpAddr=source_info_ip_s,
+    ThreatName=threat_s,
+    EndpointId=endpoint_id_g,
+    DvcType=endpoint_type_s,
```
Comment (v-admahe) [inline Solutions/SophosEP/Parsers/SophosEPEvent.txt:15]: It should be SrcDvcType. Plesae refer the  naming convention from [here](https://docs.microsoft.com/en-us/azure/sentinel/normalization-schema).

Thanks,
Aditi Maheshwari

**Review by v-admahe — CHANGES_REQUESTED** [review]

**Comment by v-admahe** [comment]
@vu-socprime The column name you provided in the parser and sample data are not matching. For example in parser you use the customer_id_g but in sample data Customerid_g is available .There are other field as well please check and make the required changes.

Thanks,
Aditi Maheshwari

File: Solutions/SophosEP/Parsers/SophosEPEvent.txt (hunk 3 above)
Comment (vu-socprime) [reply Solutions/SophosEP/Parsers/SophosEPEvent.txt]: updated

**Comment by vu-socprime** [comment]
> @vu-socprime The column name you provided in the parser and sample data are not matching. For example in parser you use the customer_id_g but in sample data Customerid_g is available .There are other field as well please check and make the required changes.
> 
> Thanks,
> Aditi Maheshwari

@v-admahe Sample data fixed.

**Review by v-admahe — CHANGES_REQUESTED** [review]

File: Sample Data/Custom/SophosEP_CL.json [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+        "user_id": "60b88e8dfd5fd40e3b6eaad2",
+        "customer_id": "84b93e3c-7299-45ef-81ff-a548fb5754a9",
+        "severity": "low",
+        "created_at": "6/15/2021, 7:29:45.191 AM",
+        "source_info_ip": "10.211.55.3",
+        "threat": "AMSI/Mimikatz-A",
+        "endpoint_id": "82912e0e-660c-4b84-8e1f-b46c2cd8b041",
+        "endpoint_type": "computer",
+        "origin": "AMSI",
+        "when": "6/15/2021, 7:29:41.837 AM",
```
[the remark this answers is not in the document]
Comment (vu-socprime) [reply Sample Data/Custom/SophosEP_CL.json]: updated

File: Sample Data/Custom/SophosEP_CL.json [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+        "SourceSystem": "RestAPI",
+        "MG": "",
+        "ManagementGroupName": "",
+        "TimeGenerated": "6/15/2021, 10:25:00.818 AM",
+        "Computer": "",
+        "RawData": "",
+        "user_id": "60b88e8dfd5fd40e3b6eaad2",
+        "customer_id": "84b93e3c-7299-45ef-81ff-a548fb5754a9",
+        "severity": "low",
+        "created_at": "6/15/2021, 7:29:45.191 AM",
```
[the remark this answers is not in the document]
Comment (vu-socprime) [reply Sample Data/Custom/SophosEP_CL.json]: updated

**Review by v-admahe — APPROVED** [review]

**Review by v-jayakal — APPROVED** [review]

**Pull request merged**